### DISCUSSION TITLE
How to filter MicroPython devices?

### DISCUSSION THREAD

**Discussion by Ma7moud3ly** [micropython/micropython#11272] [General]
Dears,
As I'm working on Android MicroPython IDE [micro-repl](https://github.com/Ma7moud3ly/micro-repl/), I need to filter the connected devices.
Here are the device details of the Raspberry Pi Pico board which is the only board I have 😅:
```
UsbDevice[Name=/dev/bus/usb/001/002,VendorId=11914,ProductId=5,Class=239,
Subclass=2,mProtocol=1,ManufacturerName=MicroPython,
ProductName=Board in FS modemVersion=1.00
```
So the question is: **ManufacturerName=MicroPython** for all MicroPython ports?
I know that I can use the vendor id, but I need to find the vendor id for all supported ports 😐

**Comment by FakeChimera01** [comment]
How i can help you? I have some micropython compatible card, how do I recover that data?

**Comment by karfas** [comment]
> So the question is: ManufacturerName=MicroPython for all MicroPython ports?
I know that I can use the vendor id, but I need to find the vendor id for all supported ports neutral_face

No. Most  of the ESP32 devices I own use a generic USB/UART converter like the CH340 or CP210x and report as such (regardless whether micropython is installed or not), with the exception of an ESP32-C3 board which reports as an "Espressif USB JTAG/serial debug unit".
Attaching the output of lsusb -v of some development boards (most likely produced in China) and an ESP32-C3 (from Olimex, Bulgaria).

[ch430.txt](https://github.com/micropython/micropython/files/11255209/ch430.txt)
[cp2104.txt](https://github.com/micropython/micropython/files/11255210/cp2104.txt)
[esp32-c3.txt](https://github.com/micropython/micropython/files/11255238/esp32-c3.txt)

**Comment by Wind-stormger** [comment]
However, if I remember correctly, most ESP32S3 and ESP32S2 devices use the USB-CDC function provided by the Integrated USB as the micropython REPL output port.

Here is the information I have obtained:
```
ESP32S2 + MicroPython firmware ,  Integrated USB 
VID = 303A
Device Vendor: Espressif Systems

ESP32S3 + MicroPython firmware ,  Integrated USB 
VID = 303A
Device Vendor: Espressif Systems

ESP32C3 + MicroPython firmware ,  Integrated USB 
VID = 303A
Device Vendor: Espressif

All CP210X UART
VID = 10C4

All CH340 UART
VID = 1A86

RP2040 + MicroPython firmware ,  Integrated USB  
VID = 2E8A
Device Vendor: MicroPython
```

I highly recommend that you first identify the VID( idVendor) to determine if it is an available device.

The VID of all Espressif equipment is 303A, but the PID is customized by the manufacturer:
https://github.com/espressif/usb-pids/blob/main/allocated-pids.txt

And during the firmware compilation of microsython, the manufacturer can also be customized, such as this board:
https://github.com/micropython/micropython/blob/b525f1c9ec8ffa9009754578932f3fad5f63026b/ports/esp32/boards/UM_FEATHERS3/sdkconfig.board#L21

@Ma7moud3ly This should help you a bit.

**Reply by Ma7moud3ly** [reply]
thank you 🙌
